File: ruby/simple_test.rb
# coding: utf-8
require 'rspec'
require 'selenium-webdriver'

describe 'DuckDuckGo search' do
  before(:each) do
    @driver = Selenium::WebDriver.for :chrome
    @wait = Selenium::WebDriver::Wait.new(:timeout => 10)
  end

  it 'should find webdriver' do
    @driver.navigate.to 'http://duckduckgo.com'

    element = @driver.find_element(:name, 'q')
    element.send_keys('webdriver')
    @driver.find_element(:id, 'search_button_homepage').click
    @wait.until { @driver.title == 'webdriver at DuckDuckGo'}
  end

  after(:each) do
    @driver.quit
  end
end

File: js/simple_test.js
var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until,
    test = require('selenium-webdriver/testing');

test.describe('Search', function() {
    var driver;

    test.before(function() {
	driver = new webdriver.Builder()
	    .forBrowser('chrome')
	    .build();
    });

    test.it('should append query to title', function() {
	driver.get('https://duckduckgo.com');
	driver.findElement(By.name('q')).sendKeys('webdriver');
	driver.findElement(By.id('search_button_homepage')).click();
	driver.wait(until.titleIs('webdriver at DuckDuckGo'), 1000);
    });

    test.after(function() {
	driver.quit();
    });
});
